src/core/annotations/qgsannotationitem.h

```diff
@@ -0,0 +1,155 @@
+/***************************************************************************
+    qgsannotationitem.h
+    ----------------
+    copyright            : (C) 2019 by Sandro Mani
+    email                : smani at sourcepole dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSANNOTATIONITEM_H
+#define QGSANNOTATIONITEM_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgscoordinatereferencesystem.h"
+#include "qgsrendercontext.h"
+#include "qgslinestring.h"
+#include "qgspolygon.h"
+
+class QgsFeedback;
+class QgsMarkerSymbol;
+class QgsLineSymbol;
+class QgsFillSymbol;
+
+/**
+ * \ingroup core
+ * Abstract base class for annotation items which are drawn with QgsAnnotationLayers.
+ *
+ * \since QGIS 3.16
+ */
+class CORE_EXPORT QgsAnnotationItem
+{
+
+#ifdef SIP_RUN
+    SIP_CONVERT_TO_SUBCLASS_CODE
+    if ( sipCpp->type() == "marker" )
```

Comment:
We don't have anything more performant than strings here?

```suggestion
    if ( sipCpp->type() == QLatin1String( "marker" ) )
```

Reply:
Again, I'm following the pattern set by symbol layers. I suppose we could use dynamic casts as the tests... but with sip I'm inclined to leave what works :laughing: